checker/src/main/java/alluxio/checker/MapReduceIntegrationChecker.java

```diff
@@ -0,0 +1,371 @@
+        while ((nextLine = curOutputFileReader.readLine()) != null) {
+          Status curOutputStatus = Status.valueOf(nextLine
+              .substring(0, nextLine.indexOf(";")).trim());
+          String curOutputAddresses = nextLine.substring(nextLine.indexOf(";") + 1).trim();
+          switch (curOutputStatus) {
+            case FAIL_TO_FIND_CLASS:
+              canFindClass = false;
+              cannotFindClassBuilder.append(" ").append(curOutputAddresses);
+              break;
+            case FAIL_TO_FIND_FS:
+              canFindFS = false;
+              cannotFindFSBuilder.append(" ").append(curOutputAddresses);
+              break;
+            default:
+              successBuilder.append(" ").append(curOutputAddresses);
+          }
+        }
+      }
+    }
+
+    String cannotFindClassAddresses = cannotFindClassBuilder.toString();
+    String cannotFindFSAddresses = cannotFindFSBuilder.toString();
+    String successAddresses = successBuilder.toString();
+
+    // Creates a file to save user-facing messages
+    File reportFile = new File("./MapReduceIntegrationReport.txt");
+    try (PrintWriter reportWriter = new PrintWriter(reportFile)) {
+      SimpleDateFormat df = new SimpleDateFormat("dd/MM/yy HH:mm:ss");
+      Date date = new Date();
+      reportWriter.printf("%n%n%n***** The integration checker ran at %s. *****%n%n",
+          df.format(date));
+
+      if (!cannotFindClassAddresses.equals("")) {
+        reportWriter.printf("Hadoop nodes of IP addresses: %s "
+            + "cannot recognize Alluxio classes.%n%n", cannotFindClassAddresses);
+      }
+      if (!cannotFindFSAddresses.equals("")) {
+        reportWriter.printf("Hadoop nodes of IP addresses: %s "
+            + "cannot recognize Alluxio filesystem.%n%n", cannotFindFSAddresses);
+      }
+      if (!successAddresses.equals("")) {
+        reportWriter.printf("Hadoop nodes of IP addresses: %s "
+            + "can recognize Alluxio filesystem.%n%n", successAddresses);
+      }
+
+      if (!canFindClass) {
+        reportWriter.println(FAIL_TO_FIND_CLASS_MESSAGE);
+        reportWriter.println(TEST_FAILED_MESSAGE);
+      } else if (!canFindFS) {
+        reportWriter.println(FAIL_TO_FIND_FS_MESSAGE);
+        reportWriter.println(TEST_FAILED_MESSAGE);
+      } else {
+        reportWriter.println(TEST_PASSED_MESSAGE);
+      }
+    }
+
+    return canFindClass ? (canFindFS ? Status.SUCCESS : Status.FAIL_TO_FIND_FS)
+        : Status.FAIL_TO_FIND_CLASS;
+  }
+
+  /**
+   * Implements MapReduce with Alluxio integration checker.
+   *
+   * @return 0 for success, 2 for unable to find Alluxio classes, 1 otherwise
+   */
+  private int run(String[] args) throws Exception {
+    Configuration conf = new Configuration();
+    String[] otherArgs =  new GenericOptionsParser(conf, args).getRemainingArgs();
+    conf.set(MRJobConfig.NUM_MAPS, otherArgs[0]);
+    conf.set("mapred.textoutputformat.separator", ";"); // Supports Hadoop 1.x
```

Comment:
do we want to support hadoop 1.x? In this class, do we only use Hadoop APIs that work for both hadoop-1.x and hadoop-2.x?

Reply:
org.apache.hadoop.mapreduce.MRJobConfig and  org.apache.hadoop.fs.FileSystem.getFileSystemClass are not supported in Hadoop 1.X, we may find ways to avoid using MRJobConfig but getFileSystemClass is so important in our design. We may focus on Hadoop 2.x